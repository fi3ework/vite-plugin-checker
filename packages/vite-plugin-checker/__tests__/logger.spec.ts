import strip from 'strip-ansi'
import { describe, expect, it } from 'vitest'

import {
  diagnosticToTerminalLog,
  normalizeEslintDiagnostic,
} from '../src/logger'
import {
  error1 as eslintError1,
  eslintResult1,
  warning1 as eslintWarning1,
} from './fixtures/eslintDiagnostic'

describe('logger', () => {
  describe('diagnosticToTerminalLog', () => {
    it('get error', () => {
      const received = strip(diagnosticToTerminalLog(eslintError1, 'ESLint'))
      expect(received).toMatchSnapshot()
    })

    it('get warning', () => {
      const received = strip(diagnosticToTerminalLog(eslintWarning1, 'ESLint'))
      expect(received).toMatchSnapshot()
    })
  })

  describe('normalizeEslintDiagnostic', () => {
    it('get multiple diagnostics', () => {
      const received = normalizeEslintDiagnostic(eslintResult1)
      expect(received).toMatchSnapshot()
    })
  })
})
