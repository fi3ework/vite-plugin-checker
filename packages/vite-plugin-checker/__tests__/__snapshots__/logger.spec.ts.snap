// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`logger > diagnosticToTerminalLog > get error 1`] = `
" ERROR(ESLint)  Unexpected var, use let or const instead.
 FILE  /Users/vite-plugin-checker/playground/eslint/src/main.ts:3:1

    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello'
      | ^^^^^^^^^^^^^^^^^^^
    4 |
    5 | const rootDom = document.querySelector('#root')!
    6 | rootDom.innerHTML = hello + text
"
`;

exports[`logger > diagnosticToTerminalLog > get warning 1`] = `
" WARNING(ESLint)  Unexpected var, use let or const instead.
 FILE  /Users/vite-plugin-checker/playground/eslint/src/main.ts:3:1

    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello'
      | ^^^^^^^^^^^^^^^^^^^
    4 |
    5 | const rootDom = document.querySelector('#root')!
    6 | rootDom.innerHTML = hello + text
"
`;

exports[`logger > normalizeEslintDiagnostic > get multiple diagnostics 1`] = `
[
  {
    "checker": "ESLint",
    "codeFrame": "  [0m [90m 1 |[39m [36mimport[39m { text } [36mfrom[39m [32m'./text'[39m
   [90m 2 |[39m
  [31m[1m>[22m[39m[90m 3 |[39m [36mvar[39m hello [33m=[39m [32m'Hello'[39m
   [90m   |[39m [31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m
   [90m 4 |[39m [36mvar[39m hello1 [33m=[39m [32m'Hello1'[39m
   [90m 5 |[39m
   [90m 6 |[39m [36mconst[39m rootDom [33m=[39m document[33m.[39mquerySelector([32m'#root'[39m)[33m![39m[0m",
    "conclusion": "",
    "id": "/Users/vite-plugin-checker/playground/eslint/src/main.ts",
    "level": 1,
    "loc": {
      "end": {
        "column": 20,
        "line": 3,
      },
      "start": {
        "column": 1,
        "line": 3,
      },
    },
    "message": "Unexpected var, use let or const instead. (no-var)",
    "stripedCodeFrame": "    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello'
      | ^^^^^^^^^^^^^^^^^^^
    4 | var hello1 = 'Hello1'
    5 |
    6 | const rootDom = document.querySelector('#root')!",
  },
  {
    "checker": "ESLint",
    "codeFrame": "  [0m [90m 2 |[39m
   [90m 3 |[39m [36mvar[39m hello [33m=[39m [32m'Hello'[39m
  [31m[1m>[22m[39m[90m 4 |[39m [36mvar[39m hello1 [33m=[39m [32m'Hello1'[39m
   [90m   |[39m [31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m
   [90m 5 |[39m
   [90m 6 |[39m [36mconst[39m rootDom [33m=[39m document[33m.[39mquerySelector([32m'#root'[39m)[33m![39m
   [90m 7 |[39m rootDom[33m.[39minnerHTML [33m=[39m hello [33m+[39m text[0m",
    "conclusion": "",
    "id": "/Users/vite-plugin-checker/playground/eslint/src/main.ts",
    "level": 1,
    "loc": {
      "end": {
        "column": 22,
        "line": 4,
      },
      "start": {
        "column": 1,
        "line": 4,
      },
    },
    "message": "Unexpected var, use let or const instead. (no-var)",
    "stripedCodeFrame": "    2 |
    3 | var hello = 'Hello'
  > 4 | var hello1 = 'Hello1'
      | ^^^^^^^^^^^^^^^^^^^^^
    5 |
    6 | const rootDom = document.querySelector('#root')!
    7 | rootDom.innerHTML = hello + text",
  },
]
`;
